pg_shard.c

```diff
@@ -782,36 +1091,130 @@ RowAndColumnFilterQuery(Query *query, List *remoteRestrictList, List *localRestr
 
 	targetList = TargetEntryList(uniqueColumnList);
 
-	filterQuery = makeNode(Query);
-	filterQuery->commandType = CMD_SELECT;
-	filterQuery->rtable = rangeTableList;
-	filterQuery->jointree = fromExpr;
-	filterQuery->targetList = targetList;
-
-	return filterQuery;
+	return targetList;
 }
 
 
 /*
  * BuildLocalQuery returns a copy of query with its quals replaced by those
- * in localRestrictList. Expects queries with a single entry in their FROM
- * list.
+ * in localRestrictList. Also, the function updates query fields that are related to
+ * aggregates if necessary. Expects queries with a single entry in their FROM list.
  */
 static Query *
-BuildLocalQuery(Query *query, List *localRestrictList)
+BuildLocalQuery(Query *query, List *localRestrictList, bool localQueryHasAggregates,
+				bool remoteQueryHasAggregates)
 {
 	Query *localQuery = copyObject(query);
 	FromExpr *joinTree = localQuery->jointree;
+	bool aggregatesPushedDown = (remoteQueryHasAggregates && !localQueryHasAggregates);
```

Comment:
Now that I see this here, I fail to understand why we sometimes say `aggregatesPushedDown` and sometimes say `remoteQueryHasAggregates`. In `BuildDistributedQuery` you test the latter but here you're testing the former. Given that there are only two possible permutations, what's the difference?